src/microbe_stage/CompoundCloudSpawner.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   Spawns compound clouds of a certain type
+/// </summary>
+public class CompoundCloudSpawner
+{
+    private readonly CompoundCloudSystem clouds;
+
+    public CompoundCloudSpawner(CompoundCloudSystem clouds)
+    {
+        this.clouds = clouds ?? throw new ArgumentException("clouds is null");
+    }
+
+    public CompoundCloudSystem GetCloudSystem()
```

Comment:
Getters should not be used. Properties with public get should be used instead. Though why is the clouds read out of this class? To me that seems a bit bad design, it should only be so that this class is given a CompoundCloudSystem object, but it isn't ever retrieved out of this class instance.